doordeck-sdk/src/androidMain/kotlin/com/doordeck/multiplatform/sdk/api/AccountResource.android.kt

```diff
@@ -52,6 +54,10 @@ actual interface AccountResource {
 
     fun registerEphemeralKeyWithSecondaryAuthenticationAsync(publicKey: ByteArray, method: TwoFactorMethod? = null): CompletableFuture<RegisterEphemeralKeyWithSecondaryAuthenticationResponse>
 
+    suspend fun registerEphemeralKeyWithSecondaryAuthenticationWithContext(method: TwoFactorMethod? = null): RegisterEphemeralKeyWithSecondaryAuthenticationResponse
```

Comment:
I have been thinking about how it can be simplified. I wouldn't want to remove the non-context methods, as they provide full control over what you are doing, and limiting it to the context may not fit all use cases (e.g., a backend application).

We could replace this:
```kotlin
    suspend fun registerEphemeralKeyRequest(publicKey: ByteArray): RegisterEphemeralKeyResponse {
        return registerEphemeralKey(publicKey)
    }

    suspend fun registerEphemeralKeyWithContextRequest(): RegisterEphemeralKeyResponse {
        val publicKey =  contextManager.getPublicKey()
            ?: throw MissingContextFieldException("Key pair is missing")
        return registerEphemeralKey(publicKey)
    }

    private suspend fun registerEphemeralKey(publicKey: ByteArray): RegisterEphemeralKeyResponse {
        val publicKeyEncoded =  publicKey.encodeByteArrayToBase64()
        return httpClient.post<RegisterEphemeralKeyResponse>(Paths.getRegisterEphemeralKeyPath()) {
            addRequestHeaders()
            setBody(RegisterEphemeralKeyRequest(publicKeyEncoded))
        }.also {
            contextManager.setUserId(it.userId)
            contextManager.setCertificateChain(it.certificateChain)
        }
    }
```

With this (In this case, using a null `publicKey` would simply cause it to use the value from the context):
```kotlin
    suspend fun registerEphemeralKeyRequest(publicKey: ByteArray? = null): RegisterEphemeralKeyResponse {
        val publicKeyEncoded = publicKey?.encodeByteArrayToBase64()
            ?: contextManager.getPublicKey()?.encodeByteArrayToBase64()
            ?: throw MissingContextFieldException("PublicKey is missing")
        return httpClient.post<RegisterEphemeralKeyResponse>(Paths.getRegisterEphemeralKeyPath()) {
            addRequestHeaders()
            setBody(RegisterEphemeralKeyRequest(publicKeyEncoded))
        }.also {
            contextManager.setUserId(it.userId)
            contextManager.setCertificateChain(it.certificateChain)
        }
    }
```

This solution fits well in all these cases, but not as much for the lock operations:

We could replace this:
```kotlin
    suspend fun unlockWithContextRequest(lockId: String, directAccessEndpoints: List<String>?) {
        val operationContext = contextManager.getOperationContext()
        val baseOperation = LockOperations.BaseOperation(
            userId = operationContext.userId,
            userCertificateChain = operationContext.userCertificateChain,
            userPrivateKey = operationContext.userPrivateKey,
            lockId = lockId
        )
        unlockRequest(LockOperations.UnlockOperation(baseOperation, directAccessEndpoints))
    }

    suspend fun unlockRequest(unlockOperation: LockOperations.UnlockOperation) {
        val operationRequest = LockOperationRequest(locked = false)
        performOperation(unlockOperation.baseOperation, operationRequest, unlockOperation.directAccessEndpoints)
    }
```

With this (In this case using null at the `userId`, `userCertificateChain` and `userPrivateKey` fields from the `BaseOperation` would simply cause it to use the values from the context):

```kotlin
    suspend fun unlockRequest(unlockOperation: LockOperations.UnlockOperation) {
        val operationRequest = LockOperationRequest(locked = false)
        val userId = unlockOperation.baseOperation.userId
            ?: contextManager.getUserId() 
            ?: throw MissingOperationContextException("User id is missing")
        val userCertificateChain = unlockOperation.baseOperation.userCertificateChain 
            ?: contextManager.getCertificateChain()
            ?: throw MissingOperationContextException("Certificate chain is missing")
        val userPrivateKey = unlockOperation.baseOperation.userPrivateKey
            ?: contextManager.getPrivateKey()
            ?: throw MissingOperationContextException("PrivateKey is missing")
        val baseOperation = LockOperations.BaseOperation(
            userId = userId,
            userCertificateChain = userCertificateChain,
            userPrivateKey = userPrivateKey,
            lockId = unlockOperation.baseOperation.lockId,
        )
        performOperation(baseOperation, operationRequest, unlockOperation.directAccessEndpoints)
    }
    }
```

We are essentially reducing the number of functions from three to one, which is not bad at all! Let me know what do you think!